skada/deep/_adversarial.py

```diff
@@ -370,6 +369,148 @@ def CDAN(
     return net
 
 
+class MDDLoss(BaseDALoss):
+    """Loss MDD.
+
+    This loss tries to minimize the disparity discrepancy between
+    the source and target domains. The discrepancy is estimated
+    through adversarial training of three networks: an encoder,
+    a task network and a discriminator.
+
+    See [35]_ for details.
+
+    Parameters
+    ----------
+    domain_criterion : torch criterion (class)
+        The criterion (loss) used to compute the
+        MDD loss. If None, a CrossEntropyLoss is used.
+    gamma : float (default=4.0)
+        Margin parameter following [35]_
+
+    References
+    ----------
+    .. [35] Yuchen Zhang et. al. Bridging Theory and Algorithm
+            for Domain Adaptation. In International Conference on
+            Machine Learning, 2019.
+    """
+
+    def __init__(self, domain_criterion, gamma=4.0):
+        super().__init__()
+        if domain_criterion is None:
+            self.domain_criterion_ = torch.nn.CrossEntropyLoss()
+        else:
+            self.domain_criterion_ = domain_criterion
+        self.gamma = gamma
+
+    def forward(
+        self,
+        y_s,
+        y_pred_s,
+        y_pred_t,
+        domain_pred_s,
+        domain_pred_t,
+        features_s,
+        features_t,
+    ):
+        """Compute the domain adaptation loss"""
+        # TODO: handle binary classification
+        # Multiclass classification
+        pseudo_label_s = torch.argmax(y_pred_s, axis=-1)
+        pseudo_label_t = torch.argmax(y_pred_t, axis=-1)
+
+        disc_loss_src = self.domain_criterion_(domain_pred_s, pseudo_label_s)
+        disc_loss_tgt = self.domain_criterion_(domain_pred_t, pseudo_label_t)
+
+        # Compute the MDD loss value
+        disc_loss = self.gamma * disc_loss_src - disc_loss_tgt
+
+        return disc_loss
+
+
+def MDD(
+    module,
+    layer_name,
+    reg=1,
+    gamma=4.0,
+    domain_classifier=None,
```

Comment:
same for domain_classifier -> disc_classifier

Reply:
Done. I also changed domain_pred to disc_pred.